homeassistant/components/roborock/number.py

```diff
@@ -88,13 +106,15 @@ def __init__(
     ) -> None:
         """Create a number entity."""
         self.entity_description = entity_description
-        super().__init__(unique_id, coordinator)
+        super().__init__(unique_id, coordinator.device_info, coordinator.api)
+
+    @property
+    def native_value(self) -> float | None:
+        """Get native value."""
+        return self.get_cache(self.entity_description.cache_key).value
 
     async def async_set_native_value(self, value: float) -> None:
         """Set number value."""
-        await self.entity_description.set_value(self, int(value))
-
-    async def async_update(self) -> None:
-        """Update number."""
-        number = await self.entity_description.get_value(self)
-        self._attr_native_value = float(number)
+        await self.entity_description.update_value(
+            self.get_cache(self.entity_description.cache_key), value
+        )
```

Comment:
It looks like entity_description.update_value is trivial enough that it should just be inlined. Is this equivalent to something like this?

```
cache = self.get_cache(self.entity_description.cache_key)
await cache.update_value([int(value)])
```